### PR TITLE
Jenkins: Add function for check state on device

### PR DESCRIPTION
**Pull request by avildema**
need stop CI it port in init state

### CODE REVIEW

File: contrib/test_jenkins.sh [hunk 1]
```diff
@@ -235,6 +235,19 @@ get_ifaddr() {
 	echo $(ip addr show ${iface} | awk '/inet /{print $2}' | awk -F '/' '{print $1}')
 }
 
+#
+# Check IB devices on state
+#
+#
+check_machine() {
+	device_list=$(ibv_devinfo -l | tail -n +2 | sed -e 's/^[ \t]*//' | head -n -1)
```
Comment (yosefe) [inline contrib/test_jenkins.sh:243]: why need sed and head? `$()` already removes extra spaces

File: contrib/test_jenkins.sh [hunk 2]
```diff
@@ -235,6 +235,19 @@ get_ifaddr() {
...
+	for ibdev in $device_list
+	do
+		port=1
+		(ibv_devinfo -d $ibdev -i $port | grep -q PORT_INIT) && { echo "$ibdev:$port have init state"; exit 1; } || true
```
Comment (yosefe) [inline contrib/test_jenkins.sh:247]:
```
if ibv_devinfo -d $ibdev -i $port | grep -q PORT_INIT
  echo ...
  exit 1
fi
```

File: contrib/test_jenkins.sh [hunk 3]
```diff
@@ -235,6 +235,19 @@ get_ifaddr() {
...
 
+#
+# Check IB devices on state
+#
+#
+check_machine() {
+	device_list=$(ibv_devinfo -l | tail -n +2 | sed -e 's/^[ \t]*//' | head -n -1)
+	for ibdev in $device_list
+	do
+		port=1
```
Comment (yosefe) [inline contrib/test_jenkins.sh:246]: `num_ports=$(ibv_devinfo -d $ibdev| awk '/phys_port_cnt:/ {print $2}')`
`for port in $(seq 1 $num_ports)`

File: contrib/test_jenkins.sh [hunk 4]
```diff
@@ -207,11 +207,36 @@ get_my_tasks() {
...
+	do
+		num_ports=$(ibv_devinfo -d $ibdev| awk '/phys_port_cnt:/ {print $2}')
+		for port in $(seq 1 $num_ports)
+		do
+			if ibv_devinfo -d $ibdev -i $port | grep -q PORT_INIT
+			then
+				echo "$ibdev:$port have init state"
+				exit 1
+			fi
+		done
```
Comment (yosefe) [inline contrib/test_jenkins.sh:239]: can we avoid code duplications?
```
check_machine() {
   init_ports=$(get_ports_in_state PORT_INIT)
   [ -n $init_ports ] && echo ... && exit
}
get_active_ib_devices() {
  get_ports_in_state PORT_ACTIVE
}
```

**Review by yosefe — COMMENTED** [review]
pls fix commit title

File: contrib/test_jenkins.sh (hunk 4 above)
Comment (avildema) [reply contrib/test_jenkins.sh:239]: good

**Review by yosefe — APPROVED** [review]

**Comment by avildema** [comment]
bot:pipe:retest

**Comment by avildema** [comment]
bot:pipe:retest

**Comment by avildema** [comment]
bot:pipe:retest

**Pull request merged**